docs/one-path-reachability-proofs.md

```diff
@@ -0,0 +1,174 @@
+Using a rule to advance a one-path reachability proof
+=====================================================
+
+This document describes how to transform a given a one-step reachability goal
+by advancing it using a given rewrite axiom.
+
+This document *does not describe* how to select an axiom to be used for
+advancing the goal.
+
+
+Problem statement
+-----------------
+
+Consider the following "one-path" reachability goal
+
+```
+∀ X . φ(X) → ◆ ∃ Y . ψ(X, Y)
+```
+where `X` and `Y` are sets of variables and ◆ denotes "strong-eventually"
+
+Consider also the axiom
+
+`∀ Z . α(Z) → ⚪ β(Z)`.
+
+where ⚪  denotes "strong-next".
+
+The steps below show how the given goal is transformed by "applying" the 
+given axiom.
+
+Unrolling "strong-eventually"
+------------------------------
+
+Let `Δ = ν f . ⚪ f` denote the formula for diverging computation.
+Then,  for any formula ψ:
+
+- `◆ ψ = ◇ ψ ∨ Δ` by the definition of "strong eventually"
+- `    = ψ ∨ ⚪ ◇ ψ ∨ Δ` by unrolling `◆`
+- `    = ψ ∨ ⚪ ◇ ψ ∨ ⚪ Δ` by unrolling `Δ`
+- `    = ψ ∨ ⚪ (◇ ψ ∨ Δ)`
+  because `⚪` commutes with `∨` 
+- `    = ψ ∨ ⚪ ◆ ψ`
+
+
+Therefore proving
+```
+∀ X . φ(X) → ◆ ∃ Y . ψ(X, Y)
+```
+is equivalent to proving
+```
+∀ X . φ(X) →  ∃ Y . ψ(X, Y) ∨ ⚪ ◆ ∃ Y . ψ(X, Y)`
+```
+
+We can now move `∃ Y . ψ(X, Y)` to the right of the implication,
+and (assuming law of excluded middle) we obtain the equivalent:
+```
+∀ X . φ(X) ∧ ¬∃ Y . ψ(X, Y) →  ⚪ ◆ ∃ Y . ψ(X, Y)`
+```
+
+Applying the axiom `∀ Z . α(Z) → ⚪ β(Z)`
+------------------------------------------
+
+In the following let `Φ(X) = φ(X) ∧ ¬∃ Y . ψ(X, Y)`
+and `Ψ(X) = ◆ ∃ Y . ψ(X, Y)`
+ 
+
+We can now split the original goal onto `∃ Y. α(Y)` to obtain
+the equivalent goal:
+
+```
+(∀ X . Φ(X) ∧ (∃ Z. α(Z)) → ⚪ Ψ(X))
+∧
+(∀ X . Φ(X) ∧ (¬ ∃ Z. α(Z)) → ⚪ Ψ(X))
+```
+
+Now we know, from the basic symbolic execution algorithm that
```

Comment:
Could you clarify which algorithm? The one in the `Algorithm for applying one rule to a pattern` document? Another one?

Also, by default, I would expect to apply the symbolic execution algorithm to φ, not to Φ. In particular, even if φ is functional, it's not obvious that Φ is functional. Could you clarify that?